### PR TITLE
Convert 5 more tests to use subtests #1520

### PR DESCRIPTION
**Pull request by spazm**
### Summary
Convert a few more tests to use subtest: `log.t, proxy.t, user_agent_unix.t , websocket_proxy.t, and hypnotoad.t`

### Motivation
requested in #1520 

### References
Issue #1520

### CODE REVIEW

File: t/mojo/hypnotoad.t [hunk 1]
```diff
@@ -151,26 +157,31 @@ while (1) {
...
+subtest "Connection did not get lost (second port)" => sub {
+  my $tx = $ua->get("http://127.0.0.1:$port2/hello");
+  ok $tx->is_finished, 'transaction is finished';
+  ok $tx->keep_alive,  'connection will be kept alive';
+  ok $tx->kept_alive,  'connection was kept alive';
+  is $tx->res->code, 200,                'right status';
+  is $tx->res->body, 'Hello Hypnotoad!', 'right content';
+};
+
+my $shared_tx;
```
Comment (kraih) [inline t/mojo/hypnotoad.t:178]: Variable outside subtest.

File: t/mojo/user_agent_unix.t [hunk 2]
```diff
@@ -20,110 +20,125 @@ use Mojo::UserAgent;
 use Mojo::Util qw(url_escape);
 use Mojolicious::Lite;
 
-# Silence
-app->log->level('fatal');
+subtest "Silence" => sub {
```
Comment (kraih) [inline t/mojo/user_agent_unix.t:23]: This again. It makes no sense.

File: t/mojo/user_agent_unix.t [hunk 3]
```diff
@@ -20,110 +20,125 @@ use Mojo::UserAgent;
...
+
+  websocket '/echo' => sub {
+    my $c = shift;
+    $c->on(message => sub { shift->send($c->req->url->to_abs->host . ': ' . shift)->finish });
+  };
+};
+
+my $test = $dir->child('test.sock');
+my $encoded = url_escape "$test";
+my $daemon;
```
Comment (kraih) [inline t/mojo/user_agent_unix.t:48]: Variables outside of subtest.

**Review by kraih — CHANGES_REQUESTED** [review]
Some tests are fine and could be merged, but others have scoping issues and require more attention.

**Comment by kraih** [comment]
I'm sorry, but i've decided to close these PRs, because they contain too many converted tests at once. And we don't have the review capacity to re-review the whole thing after every scoping issue fix.

**Comment by spazm** [comment]
Rebased away the tests that did not convert cleanly.  The PR is updated with 3 files that require only minimal non-whitespace changes.  Hopefully this will be less of a review burden.

I apologize for the state of this PR last night, I had no idea you'd come read it so quickly! (which is awesome!)

**Comment by spazm** [comment]
`log.t` was already fixed in another branch.  removed and rebase to master.

**Pull request closed without merging**